Neutral JavaScript runtime support (Deno, Bun, etc)

Hey there @tj,

First of all, big fan of `commander`; I've used it for a long time, so, longtime user first time contributor 😁. I have a patch for `commander` which makes it work neutrally on all JS runtimes; it was already working fine on Bun and Node, but with the following two small changes:

- Prefixing Node API imports with `node:*`
- Importing `process` into an explicit symbol

... it will now run smoothly on Deno as well.

For context: I'm trying to ship a tool downstream, [`hashlock`](https://github.com/sgammon/hashlock), which uses commander, but I want it to be able to run on any JS runtime.

**Distribution diff:**
```diff
diff --git a/node_modules/commander/lib/command.js b/node_modules/commander/lib/command.js
index 5b16e60..e1ddedb 100644
--- a/node_modules/commander/lib/command.js
+++ b/node_modules/commander/lib/command.js
@@ -1,8 +1,8 @@
-const EventEmitter = require('events').EventEmitter;
-const childProcess = require('child_process');
-const path = require('path');
-const fs = require('fs');
-const process = require('process');
+const EventEmitter = require('node:events').EventEmitter;
+const childProcess = require('node:child_process');
+const path = require('node:path');
+const fs = require('node:fs');
+const process = require('node:process');
 
 const { Argument, humanReadableArgName } = require('./argument.js');
 const { CommanderError } = require('./error.js');
```

PR incoming shortly.

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

I get multiple errors when I try the build. (On `main` branch.) The first error is for `glob`  which is a little interesting given it is one of the files which would get patched. The build also fails after doing a plain install with the patches applied.

```console
% bun --version
1.0.36
% pnpm --version
8.15.5
% pnpm run build

> hashlock@1.0.3 build /Users/john/Documents/Sandpits/commander/issues/2169/hashlock
> bun ./scripts/build.mjs && bun run build:standalone && bun run build:types && cp -fv ./dist/index.d.ts ./dist/index.d.mts

Building 'verify-hashes'...
- Building 'hashlock' (lib, cjs)...
- Building 'hashlock' (CLI, esm)...
- Building 'hashlock' (action)...
✘ [ERROR] Could not resolve "glob"

    src/main.ts:19:21:
      19 │ import { glob } from 'glob'
         ╵                      ~~~~~~

  The Yarn Plug'n'Play manifest forbids importing "glob" here because it's not listed as a
  dependency of this package:
...
```

> But in any case the CLI built fine in the log you have provided.

The `Building` lines are from before the async builds, and all the builds fail (to be clear, I did truncate the log). No files are being generated in `.dist/`, and in particular not `./dist/cli.mjs`.

I tried downloading a build package to examine the build files, but got an error for that too and didn't pursue further:
```console
% npm install hashlock
npm ERR! code 127
npm ERR! path /Users/john/Documents/Sandpits/commander/issues/2169/bin/node_modules/hashlock
npm ERR! command failed
npm ERR! command sh -c patch-package
npm ERR! sh: patch-package: command not found
```

--- a/lib/command.js
+++ b/lib/command.js
@@ -1,8 +1,8 @@
-const EventEmitter = require('events').EventEmitter;
-const childProcess = require('child_process');
-const path = require('path');
-const fs = require('fs');
-const process = require('process');
+const EventEmitter = require('node:events').EventEmitter;
+const childProcess = require('node:child_process');
+const path = require('node:path');
+const fs = require('node:fs');
+const process = require('node:process');
 
 const { Argument, humanReadableArgName } = require('./argument.js');
 const { CommanderError } = require('./error.js');

